src/toil/cwl/cwltoil.py

```diff
@@ -937,35 +1539,64 @@ def _collectDirEntries(
                 for dir_entry in _collectDirEntries(sub_obj):
                     yield dir_entry
 
+    # This is all the CWL File and Directory objects we need to export.
     jobfiles = list(_collectDirEntries(cwljob))
-    pm = ToilPathMapper(jobfiles, "", outdir, separateDirs=False, stage_listing=True)
-    for _, p in pm.items():
-        if p.staged:
-            if destBucket and p.type in ["File", "CreateFile"]:
-                # Directories don't need to be created if we're exporting to a bucket
-                baseName = p.target[len(outdir) :]
-                local_file_path = p.resolved[len("file://") :]
 
-                if (
-                    p.type == "CreateFile"
-                ):  # TODO: CreateFile for buckets is not under testing
-                    local_file_path = os.path.join(
-                        file_store.getLocalTempDir(), baseName
-                    )
-                    with open(local_file_path, "wb") as n:
-                        n.write(p.resolved.encode("utf-8"))
+    # TODO: I think we need to recursively populate listings here!
```

Comment:
Is this not done with the next `pm = ToilPathMapper`call?  Or old listings are not up-to-date?

Reply:
I think we don't actually need to do it after all. The path mapper and whatever stages using it is clever enough to work even without content listings.

Reply:
I think we're using our own custom code that just makes a directory to "stage" the directory, instead of using e.g. ToilFsAccess to list its files and copy them. Seems like it might be wrong, but also there's no conformance test here that we're failing.

Reply:
It's not that listings won't be up to date so much as it's that they might not exist or be recursive; the child Directory and File objects are only sometimes created at the CWL level, but all the contents are supposed to come along all the time.